examples/search_examples/public/search/app.tsx

```diff
@@ -304,14 +317,36 @@ export const SearchExamplesApp = ({
 
       setRequest(searchSource.getSearchRequestBody());
       const abortController = new AbortController();
+      const inspector: IInspectorInfo = {
+        adapter: new RequestAdapter(),
+        title: 'Example App Inspector!',
+        id: 'greatest-example-app-inspector',
+        description: 'Use the `description` field for more info about the inspector.',
+      };
       setAbortController(abortController);
       setIsLoading(true);
-      const { rawResponse: res } = await lastValueFrom(
-        searchSource.fetch$({ abortSignal: abortController.signal })
+      const result = await lastValueFrom(
+        searchSource.fetch$({
+          abortSignal: abortController.signal,
+          disableShardFailureWarning: !warningsShown,
+          inspector,
+        })
       );
-      setRawResponse(res);
+      setRawResponse(result.rawResponse);
+
+      const warnings = inspector.adapter?.extractWarnings();
+      if (warnings) {
+        setSearchSourceWarnings(warnings);
+
+        if (warningsShown) {
+          data.search.showWarnings({
+            title: 'ResponseWarning',
```

Comment:
shouldn't we rather show the actual warnings returned from inspector ?

Reply:
ccd5a3a80ee